### PR TITLE
Seq[Property] combine extension method

### PR DESCRIPTION
**Pull request by FreexD**
Closes https://github.com/UdashFramework/udash-core/issues/189

### CODE REVIEW

File: core/shared/src/main/scala/io/udash/properties/PropertySeqCombinedReadableSeqProperty.scala [hunk 1]
```diff
@@ -0,0 +1,37 @@
...
+private[properties] final class PropertySeqCombinedReadableSeqProperty[A](value: ISeq[Property[A]])
+  extends AbstractReadableSeqProperty[A, ReadableProperty[A]] {
+
+  override val id: PropertyId = PropertyCreator.newID()
+  override protected[properties] val parent: ReadableProperty[_] = null
+
+  private val children = CrossCollections.createArray[ReadableProperty[A]]
+
+  value.foreach(property => {
+    children.+=(property.transform(value => value))
```
Comment (Starzu) [inline core/shared/src/main/scala/io/udash/properties/PropertySeqCombinedReadableSeqProperty.scala]: This type of transform is not a good way to ensure that no one can access mutable API with pattern matching. I know that there is no better way now, but it's time to provide it. I think method `_.readonly` returning a minimal wrapper around the property will be much better. (Ping @ddworak)

File: core/shared/src/main/scala/io/udash/properties/PropertySeqCombinedReadableSeqProperty.scala [hunk 2]
```diff
@@ -0,0 +1,37 @@
...
+    property.listen(_ => valueChanged())
```
Comment (Starzu) [inline core/shared/src/main/scala/io/udash/properties/PropertySeqCombinedReadableSeqProperty.scala]: This implementation does not allow GC to clean up this property, when it's no longer needed, but the origin `property` is still in use. Take a look at `transform` implementation from `ReadableProperty`.

File: core/shared/src/main/scala/io/udash/properties/PropertySeqCombinedReadableSeqProperty.scala [hunk 3]
```diff
@@ -0,0 +1,37 @@
+package io.udash.properties
+
+import com.avsystem.commons.ISeq
+import io.udash.properties.Properties.{Patch, Property, ReadableProperty}
+import io.udash.properties.seq.AbstractReadableSeqProperty
+import io.udash.utils.Registration
+
+private[properties] final class PropertySeqCombinedReadableSeqProperty[A](value: ISeq[Property[A]])
```
Comment (Starzu) [inline core/shared/src/main/scala/io/udash/properties/PropertySeqCombinedReadableSeqProperty.scala]: I think that `ISeq[ReadableProperty[A]]` is enough.

File: core/shared/src/main/scala/io/udash/properties/PropertySeqCombinedReadableSeqProperty.scala (hunk 1 above)
Comment (ddworak) [reply core/shared/src/main/scala/io/udash/properties/PropertySeqCombinedReadableSeqProperty.scala]: True, one can always pattern match on the property at runtime and find out it has some nice mutable features.

Comment (Starzu) [reply core/shared/src/main/scala/io/udash/properties/PropertySeqCombinedReadableSeqProperty.scala]: #196

**Pull request merged**